src/ERC7540Vault.sol

```diff
@@ -136,10 +136,10 @@ contract ERC7540Vault is Auth, IERC7540Vault {
         // the sender is the owner, to bypass the allowance check
         address sender = isOperator[owner][msg.sender] ? owner : msg.sender;
 
-        require(
-            AuthTransferLike(share).authTransferFrom(sender, owner, address(escrow), shares),
-            "ERC7540Vault/transfer-failed"
-        );
+        try AuthTransferLike(share).authTransferFrom(sender, owner, address(escrow), shares) returns (bool success) {}
+        catch {
+            require(IERC20(share).transferFrom(owner, address(escrow), shares), "ERC7540Vault/transfer-from-failed");
```

Comment:
```suggestion            
            require(SafeTransferLib.safeTransferFrom(share, owner, address(escrow), shares), "ERC7540Vault/transfer-from-failed");
```

Reply:
Not required because the share is a trusted token, always our own implementation